watches/util/ESClientProducer.py

```diff
@@ -0,0 +1,37 @@
+"""Provider of Elasticsearch client."""
+
+from elasticsearch import Elasticsearch
+
+def create_client(options):
+    """Produces a new Elasticsearch client according to provided options."""
+
+    user_kwargs = {}
+
+    if "--url" in options:
+        user_kwargs.update({
+            'hosts': options["--url"]
+        })
+
+    # "true" is not translated to native True value ?
+    if "--sniff" in options and "true" == options["--sniff"]:
+        user_kwargs.update({
+            # sniff before doing anything
+            "sniff_on_start": True,
+            # refresh nodes after a node fails to respond
+            "sniff_on_connection_fail": True,
+            # and also every 60 seconds
+            "sniffer_timeout": 60
+        })
+
+    # We can test just for --cacert because all the three options are required
+    # if at least one of them is provided.
```

Comment:
Right, but I talking about something different here :sunglasses:

As of writing we define the CLI options (via [docopt](http://docopt.org/)) to include the following pattern:

    [(--cacert=CACERT --cert=CERT --key=KEY)]

Which means that these options are optional, BUT if any of them is used then ALL the other are required too, or the options parsing fails. So in the code level it is enough the test for presence of just a single option and we are guarantied that if any of them is present we can safely assume the others are set as well.

So my comment is about the code check only, not about which security tokens are required for successful communication (ATM _password_ and _username_ options are not supported and we should probably open a ticket for it).